crates/flightrepl/src/lib.rs

```diff
@@ -545,77 +686,16 @@ fn lines_need_truncation(lines: &[&str]) -> bool {
-            "Internal Error",
-            "An unexpected internal error occurred. Execute '.error' for details.".to_string(),
-        ),
-        Code::InvalidArgument | Code::AlreadyExists | Code::NotFound | Code::Unavailable => {
-            let message = err.message();
-            let lines = message.split('\n').collect::<Vec<_>>();
-            let truncate = lines_need_truncation(&lines);
-
-            let first_line = lines.first().unwrap_or(&message);
-            match (truncate, lines.len() > 1) {
-                (true, true) => {
-                    // truncating due to length, and multiple error lines
-                    (
-                        "Query Error",
-                        format!(
-                            "{first_line}\nThis error message has been truncated.\nFor the full error message, execute `.error`."
-                        ),
-                    )
-                }
-                (true, false) => {
-                    // truncating due to length, but only one line
-                    ("Query Error", "Failed to execute query.\nThis error message has been truncated.\nFor the full error message, execute `.error`.".to_string())
-                }
-                _ => ("Query Error", message.to_string()),
-            }
-        }
-        Code::Cancelled => (
-            "Cancelled",
-            "The operation was cancelled before completion.".to_string(),
-        ),
-        Code::Aborted => (
-            "Aborted",
-            "The operation was aborted before completion.".to_string(),
-        ),
-        Code::DeadlineExceeded => (
-            "Timeout Error",
-            "The operation could not complete within the allowed time limit.".to_string(),
-        ),
-        Code::Unauthenticated => (
-            "Authentication Error",
-            "Access denied. Invalid credentials.".to_string(),
-        ),
-        Code::PermissionDenied => (
-            "Authorization Error",
-            "Access denied. Insufficient permisions to complete the request.".to_string(),
-        ),
-        Code::ResourceExhausted => (
-            "Resource Limit Exceeded",
-            "The operation could not be completed because the server resources are exhausted."
-                .to_string(),
-        ),
-        Code::Unimplemented => (
-            "Unsupported Operation",
-            "The query could not be completed because the requested operation is not supported."
-                .to_string(),
-        ),
-        Code::OutOfRange => (
-            "Result Limit Exceeded",
-            "The query result exceeds allowable limits. Consider using a `limit` clause."
-                .to_string(),
-        ),
-    };
-
-    println!(
-        "{} {user_err_msg}",
-        Colour::Red.paint(format!("{error_type}:"))
-    );
+fn is_table_not_found_error(err: &Status) -> (bool, String) {
+    let message = err.message();
+    if message.contains("table '") && message.contains("' not found") {
```

Comment:
We shouldn't be special casing this. We should have better logic for categorizing errors between user, dependency, and system, etc. and always showing all user errors directly into the REPL.

@phillipleblanc @peasee

Reply:
The tonic status error roughly looks like this

```rust
FlightError::Tonic(tonic::Status{
    code: Code,
    message: String,
    details: Bytes,
    metadata: MetadataMap,
    source: Option<Arc<dyn Error + Send + Sync + 'static>>
})
```

We could utilise the `MetadataMap` with spice.ai specific properties. This would resolve it, but would require some refactoring of our flight service.

Reply:
The REPL currently decides whether something is a user error or an internal error based on the `code`. A table not existing should be an INVALID_ARGUMENT error (https://grpc.io/docs/guides/status-codes/) which the REPL already knows how to display.

The underlying issue then is that we aren't returning INVALID_ARGUMENT for tables not existing from the Spice flight service.